pkg/kubeinteraction/resultsannotation.go

```diff
@@ -0,0 +1,45 @@
+package kubeinteraction
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/params/info"
+	tektonv1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+)
+
+const (
+	resGroupName            = "results.tekton.dev"
+	recordSummaryAnnotation = "/recordSummaryAnnotations"
+)
+
+type ResultAnnotation struct {
+	Repo          string `json:"repo"`
+	Commit        string `json:"commit"`
+	EventType     string `json:"eventType"`
+	PullRequestID int    `json:"pull_request-id,omitempty"`
+}
+
+// Add annotation to PipelineRuns produced by PaC for TektonResults
+// to capture data for summary and record.
+func AddResultsAnnotation(event *info.Event, pipelineRun *tektonv1.PipelineRun) error {
+	if event == nil {
```

Comment:
I dont think we need this check here, because if event is nil, it will break before coming to this point, better add it in the function `AddLabelsAndAnnotations`

Reply:
sure, I'm working on addressing it.